Recast/Source/RecastRasterization.cpp

```diff
@@ -23,433 +23,532 @@
 #include "RecastAlloc.h"
 #include "RecastAssert.h"
 
-inline bool overlapBounds(const float* amin, const float* amax, const float* bmin, const float* bmax)
+/// Check whether two bounding boxes overlap
+///
+/// @param[in]	aMin	Min axis extents of bounding box A
+/// @param[in]	aMax	Max axis extents of bounding box A
+/// @param[in]	bMin	Min axis extents of bounding box B
+/// @param[in]	bMax	Max axis extents of bounding box B
+/// @returns true if the two bounding boxes overlap.  False otherwise.
+static bool overlapBounds(const float* aMin, const float* aMax, const float* bMin, const float* bMax)
 {
-	bool overlap = true;
```

Comment:
This came from a project where it lead to branchless test.

Using `const float*` leads to a lot of branches, but a version using simple `vec3` struct gets vectorized:
https://godbolt.org/z/oaKPMqW7x

I think that could lead a good savings even if used just internally.

Reply:
Oh nice! I was hoping to do something like this already for clarity but it makes a ton of sense that it'd also improve perf since it gives much more information to the optimizer and changes things to value semantics, which eliminates the possibility of pointer shadowing.  Definitely an improvement we should make. 💯

Reply:
I'll do some more profiling but given that the original code here was branchless it's probably better to keep that over this version.

Reply:
The changed are that the original code was either using `Vec3` of sorts or `float[3]` which would allow the compiler to vectorize it. I think your version has should be better with `float*`